### ISSUE TITLE
hotkeys are confusing

### ISSUE THREAD

**Issue by kianzarrin** [0xd4d/dnSpy#1424]
I find it confusing that clicking on anything does to its declaration.
I think dnSpy needs the following improvements:
- simple click should not follow to declaration.  [modifier keys]+click should do that instead.
- alt-clicking in the assembly explorer follows to definition. There is no way I can click on stuff in the assembly explorer without following to definition.
- There is no way to navigate to current method in the assembly explorer (dot peek has such a shortcut).  Auto nevigate follows only to the current class. 
- Assembly explorer should show (ro) for reflection only assemblies. This would help with Cities skylines debugging a lot.

**Comment by CreateAndInject** [comment]
>simple click should not follow to declaration. [modifier keys]+click should do that instead.
alt-clicking in the assembly explorer follows to definition. There is no way I can click on stuff in the assembly explorer without following to definition.

Click in `assembly explorer` or `code window`? `assembly explorer` is the left `TreeView`.

>There is no way to navigate to current method in the assembly explorer (dot peek has such a shortcut). Auto nevigate follows only to the current class.

Just disable Options->Environment->Decompile a member's enclosing class instead of the class member.
Everty time I reinstall OS, I'll disable this option, I don't know why it should be default option.

>Assembly explorer should show (ro) for reflection only assemblies. This would help with Cities skylines debugging a lot.

`(ro)` ??? I can't understand you.

**Comment by kianzarrin** [comment]
> Click in assembly explorer or code window? assembly explorer is the left TreeView.

Both. But in the code window at least you have the option of pressing alt to avoid following things. in the Tree View you can't even do that. So lets say I want to click on class name and then use arrows to go up and down or navigating toward the class I want by pressing the letter they start with (like I can do in ILSpy).

> Just disable Options->Environment->Decompile a member's enclosing class instead of the class member.

Wow this is owesome!

> I don't know why it should be default option

I totally agree. I also find it hard to find. the sentence has a complex grammar and requires a lot of focus to understand. considering that there are quite a few options users tend to ignore such information. Also a bit of tool-tip would help because it is not clear that de-compiling only one member would mean it will open up in the assembly explorer. In dot peek you can de-compile the enclosing class and go to the particular member in the assembly explorer.

> (ro) ??? I can't understand you.

for get about the (ro) . When assembly is loaded as reflection only, it should have an indication (eg `AssemblyName (reflection only)` )

**Comment by CreateAndInject** [comment]
>But in the code window at least you have the option of pressing alt to avoid following things.

Before today, when I want to copy Type/Member name, I must carefully press left mouse key in the edge of the name and then move my mouse to select the name and then copy, but today I know, I can use Alt+DoubleClick to select and copy, dnSpy doesn't navigate if `Alt` is pressed.
So, click to navigate it's not a problem for me now.

> in the Tree View you can't even do that.

Do you mean dnSpy shoud do thing if someone click the node of TreeView, dnSpy shouldn't decompile the node you click?
How strange you are! Any decompiler I know will decompile the node when you click it.

**Comment by Taschenbuch** [comment]
>     * simple click should not follow to declaration.  [modifier keys]+click should do that instead.

I agree, it should be have more like visual studio for people to easier get used to it. 
A simply click should not go to definition. CTRL + CLICK should do it. I didnt know this "press alt key" trick either. 

>     * alt-clicking in the assembly explorer follows to definition. There is no way I can click on stuff in the assembly explorer without following to definition.

No, i think this should be handled like visual studio as well:
- a name space in the assembly explorer is like a folder in visual studio. Thus click on a namespace should select it. double click on a namespace should expand the tree below it
- click on a class in the assembly explorer should select it, double click on the class should show the code of it. But it should NOT expand the tree to show all the properties etc. Right now i help myself by unchecking in the options "open node with single click" (dont know if that is correctly translated, because i can not switch to the dnspy english version)

Though it might be tricky to implement those different behaviors for a tree.
